src/Razor/test/Microsoft.CodeAnalysis.Razor.Workspaces.Test.Common/TestDocumentFilePathProvider.cs

```diff
@@ -0,0 +1,16 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT license. See License.txt in the project root for license information.
+
+using Microsoft.CodeAnalysis.Razor.Workspaces;
+
+namespace Microsoft.AspNetCore.Razor.LanguageServer;
+
+internal class TestDocumentFilePathProvider : DocumentFilePathProvider
```

Comment:
In this case, I have no problem with that, but I will say I've been annoyed by that pattern before because some of the base classes construct structures for testing, like ProjectSnapshotManagers, that contain pretty specific scenarios for specific tests, so I think there is some nuance to the balance here.